### DISCUSSION TITLE
Upon Session Expiration automatically redirect to login page

### DISCUSSION THREAD

**Discussion by deeztek** [authelia/authelia#5016] [Q&A] [answered]
Is there a way for authelia to automatically redirect a user back to the login page after the session expires without the user having to refresh the page or at the very least display a message that they have been logged out thus hiding the previously displayed information? The reason I'm asking is because we have an application that's subject to NIST 800-53 controls that's behind Authelia.

Thanks

**Comment by james-d-elliott** [comment, marked as answer]
Authelia will automatically redirect users when their session has ended provided there is a request. The issue is there must be something within the third-party application which makes a request and follows the redirection itself. You could do this with javascript which periodically when the user has not changed pages in a while which checks the user can access a healthcheck page  of the app in question using the fetch API in a way that follows redirects.
